api/db/knex-database-connection.js

```diff
@@ -42,23 +43,21 @@ try {
 const knexConfig = knexConfigs[environment];
 const knex = require('knex')(knexConfig);
 
-const queries = new Map();
-
 knex.on('query', function(data) {
-  if (logging.enableLogKnexQueriesWithCorrelationId) {
-    const queryStartedTime = new Date();
-    queries.set(data.__knexQueryUid, queryStartedTime);
-    addPositionToQuerieAndIncrementQueriesCounter(data.__knexQueryUid);
+  if (logging.enableLogKnexQueriesWithCorrelationId && doesStoreContainsRequest()) {
+    const store = asyncLocalStorage.getStore();
+    const queryStartedTime = performance.now();
+    store.knexQueriesUUIDs[data.__knexQueryUid] = queryStartedTime;
   }
```

Comment:
```suggestion
  if (logging.enableLogKnexQueries) {
    monitoringTools.setInContext(`knexQueryStartTimes.${data.__knexQueryUid}`, performance.now());
  }
```

`monitoring-tools.js`: 

```js
function setInContext(path, value) {
  const store = asyncLocalStorage.getStore();
  if (!store) return;
  _.set(store, path, value);
}
```